snapcraft/_schema.py

```diff
@@ -41,10 +41,18 @@ def __init__(self, snapcraft_yaml=None):
         self._load_schema()
 
     @property
+    def schema(self):
+        """Return all schema properties."""
+
+        return self._schema['properties'].copy()
+
+    @property
     def part_schema(self):
-        sub = self._schema['properties']['parts']['patternProperties']
+        """Return part-specific schema properties."""
+
+        sub = self.schema['parts']['patternProperties']
         properties = sub['^(?!plugins$)[a-z0-9][a-z0-9+-]*$']['properties']
-        return properties.copy()
```

Comment:
Is removing `.copy()` here really needed? There might be side effects we don't know about and I know I added it for a reason.